shell/core/plugins.js

```diff
@@ -182,7 +182,11 @@ export default function({
           console.error('Error removing plugin', e); // eslint-disable-line no-console
         }
 
-        delete plugins[plugin.name];
```

Comment:
I think deleting the member of plugins by `plugin.name` may be not accurate. After apply this change, the issue seems to be solved.

@richard-cox Please take a re-review when you are free, thanks!

Reply:
cc @WuJun2016 thanks!